schemas/dlhub.json

```diff
@@ -77,8 +86,13 @@
         "test": {
             "type": "boolean",
             "description": "Internal use. Whether DLHub is actually going to do anything other than validate the request"
+        },
+        "type": {
+            "type": "string",
+            "description": "Type of this artifact.",
+            "enum": ["servable", "dataset", "pipeline"]
         }
     },
-    "required": ["version", "name"],
+    "required": ["version", "name", "type"],
```

Comment:
Shouldn't all the identifying information be required? `publication_date` and `owner` are missing from the list. I think it would be useful to guarantee that all `dlhub` blocks have this necessary information.

I would also propose that `id` and `test` be required, if they're essential to entry processing.

Reply:
Good point. I currently left them out because I was looking to define the required types needed on submission, which are fewer than those required in the database (e.g., owner gets added after submission by the server). 

Would you be OK with leaving it this way? Or, do you think the schemas need a definition of what to expect from the server?

Reply:
I may simply be thinking of the schema differently than you are. I see the DLHub schema as a guarantee of what is in a valid DLHub entry, which I can expect from the server. Otherwise, it's more difficult to tell what information will be in the DLHub search index (and database, if that's user-accessible). You seem to be using the schema as both the Search schema and the submission schema, which is a different take.

I do understand needing a schema to validate a submission, although in my experience (with MDF Connect) it's easier to have a separate "submission schema" as opposed to making one schema fit both the submission document and the final processed document, both because the required properties are easier to manage and it's easier to communicate to a user what they can and cannot submit (I assume a user can't set the `id`, for example).

Reply:
I think creating separate "submission" and "database" schemas would be helpful, but am going to pass on making it for now because the only differences are in the required fields of the DLHub block.

Reply:
I.e., I don't think separating the two is yet worth the additional complexity

Reply:
That seems reasonable to me.